lib/vrl/cli/src/repl.rs

```diff
@@ -158,21 +169,40 @@ fn resolve(
 struct Repl {
     highlighter: MatchingBracketHighlighter,
     validator: MatchingBracketValidator,
-    hinter: HistoryHinter,
+    history_hinter: HistoryHinter,
     colored_prompt: String,
+    hints: Vec<String>,
 }
 
 impl Repl {
     fn new() -> Self {
         Self {
             highlighter: MatchingBracketHighlighter::new(),
-            hinter: HistoryHinter {},
+            history_hinter: HistoryHinter {},
             colored_prompt: "$ ".to_owned(),
             validator: MatchingBracketValidator::new(),
+            hints: initial_hints(),
         }
     }
 }
 
+fn initial_hints() -> Vec<String> {
+    let mut hints: Vec<String> = Vec::new();
```

Comment:
It doesn't matter in this case, but for future reference; since we know the length of `stdlib::all()`, you can pre-allocate the correct number of bytes by using `Vec::with_capacity(stdlib::all().len())`. This improves performance (i.e. less resizing of the allocated memory block).

Reply:
Yup, no good reason not to make this a habit 👍🏼